src/cli/cmd/login/mod.rs

```diff
@@ -209,6 +211,11 @@ impl LoginSubcommand {
                 "failed to update netrc via determinatenixd, falling back to local-file approach"
             );
 
+            // check if user is root or not
+            if !nix::unistd::Uid::effective().is_root() {
+                return Err(eyre!("`fh login` is attempting to update a file owned by root, please re-run the same command, prefixed with `sudo -i`."))
+            }
+
```

Comment:
I'm still missing something, this outputs the backtrace message too, even though it's just un-Try'd in this function's caller.